go_utils/tests/photo_download_test.py

```diff
@@ -69,48 +71,168 @@
     [
         (
             "go_utils/tests/sample_data/mhm_small.csv",
             get_mhm_download_targets,
-            desired_mhm_names,
+            mhm_name_fields,
+            "",
+            full_mhm_names,
         ),
         (
             "go_utils/tests/sample_data/lc_small.csv",
             get_lc_download_targets,
-            desired_lc_names,
+            lc_name_fields,
+            "",
+            full_lc_names,
         ),
     ],
 )
-def test_naming(input_file, func, desired):
+def test_all_field_naming(input_file, func, included_fields, additional_info, desired):
     df = pd.read_csv(input_file)
     convert_dates_to_datetime(df)
 
-    targets = func(df, "")
-    success = True
+    targets = func(
+        df,
+        directory="",
+        include_in_name=included_fields,
+        additional_name_stem=additional_info,
+    )
     output_filenames = [target[2] for target in targets]
+    assert len(output_filenames) == len(
+        desired
+    ), "desired len does not equal output_filenames len"
+
+    # Check set equality of desired/expected and output_filenames
     for filename in desired:
-        if filename not in output_filenames:  # pragma: no cover
-            success = False
-            print(filename)
+        assert filename in output_filenames, (
+            filename + " from desired is not in output_filenames."
+        )
 
     for filename in output_filenames:
-        if filename not in desired:  # pragma: no cover
-            success = False
-            print(filename)
+        assert filename in output_filenames, (
+            filename + " from output_filenames is not in desired."
+        )
+
+
+@pytest.mark.photodownload
+@pytest.mark.parametrize(
+    "input_file, func, included_fields, desired",
+    [
+        (
+            "go_utils/tests/sample_data/mhm_small.csv",
+            get_mhm_download_targets,
+            [],
+            [(re.sub(r"mhm\_.*\_", "mhm_", x)) for x in full_mhm_names],
+        ),
+        (
+            "go_utils/tests/sample_data/lc_small.csv",
+            get_lc_download_targets,
+            [],
+            [(re.sub(r"lc\_.*\_", "lc_", x)) for x in full_lc_names],
+        ),
+    ],
+)
+def test_no_field_naming(input_file, func, included_fields, desired):
+    test_all_field_naming(input_file, func, included_fields, "", desired)
+
+
+@pytest.mark.photodownload
+@pytest.mark.parametrize(
+    "input_file, func, included_fields, additional_info, desired",
+    [
+        (
+            "go_utils/tests/sample_data/mhm_small.csv",
+            get_mhm_download_targets,
+            [],
+            "ADDITIONAL",
+            [(re.sub(r"mhm\_.*\_", "mhm_ADDITIONAL_", x)) for x in full_mhm_names],
+        ),
+        (
+            "go_utils/tests/sample_data/lc_small.csv",
+            get_lc_download_targets,
+            [],
+            "ADDITIONAL",
+            [(re.sub(r"lc\_.*\_", "lc_ADDITIONAL_", x)) for x in full_lc_names],
```

Comment:
If additional info just adds a string that doesn't necessarily relate with each individual entry, is it worth adding?

Reply:
I think so -- For example, perhaps a User wants to download 2 mutually exclusive sets of images. The additional Information parameter could help the User organize those 2 sets.

Reply:
Ah that makes sense